docs: add documentation for the new type(slice of any) in Go (#474)
awsdocs/aws-cdk-guide: v2/guide/work-with-cdk-go.adoc

jsii.Strings("One", "Two", "Three")
----

[#go-any-slice]
=== Working with any slice

> In [.noloc]`Go`, certain L1 constructs expect properties defined as a slice of any (`*[]interface{}`). This is because these properties can accept multiple types of values (similar to union types in TypeScript). The Go bindings represent this flexibility using a slice of any.

Reviewer: ```suggestion
Certain constructs expect properties that are a list of multiple types (union types in TypeScript). In [.noloc]`Go`, these are as a slice of any (`*[]any`). The `any` ensures that the compiler will allow the assignment of different types there. Refer to the documentation to find out what the allowed types are.
```

The {aws} CDK automatically compiles your app before running it. However, it can be useful to build your app manually to check for errors and to run tests. You can do this by issuing  `go build` at a command prompt while in your project's root directory.

Run any tests you've written by running `go test` at a command prompt.

[#go-troubleshooting]
== Troubleshooting

If you encounter a compiler error like the following:

[source,none]
----
Cannot use 'jsii.Strings("arn:aws:s3:::bucket1", "arn:aws:s3:::bucket2")' (type *[]*string) as the type *[]interface{}
----

> …it means string slices were passed directly to a property expecting a slice of any. To solve this, replace `jsii.Strings()` with `jsii.AnyStrings()`. Refer to [this GitHub issue]() for more context and additional solutions.

Reviewer 2: ```suggestion
…it means string slices were passed directly to a property expecting a slice of any. To solve this, replace `jsii.Strings()` with `jsii.AnyStrings()`.
```

The {aws} CDK automatically compiles your app before running it. However, it can be useful to build your app manually to check for errors and to run tests. You can do this by issuing  `go build` at a command prompt while in your project's root directory.

Run any tests you've written by running `go test` at a command prompt.

[#go-troubleshooting]
== Troubleshooting

If you encounter an error like the following when running `cdk deploy` or `cdk synth`:

[source,none]
----

> panic: Passed to parameter props of new aws-cdk-lib.aws_iam.CfnUser: Unable to deserialize value as aws-cdk-lib.aws_iam.CfnUserProps

Reviewer: Error is different.

        ├── 🛑 Failing value is an object
        │      { '$jsii.struct': [Object] }
        ╰── 🔍 Failure reason(s):
            ╰─ Key 'managedPolicyArns': Unable to deserialize value as aws-cdk-lib.IResolvable | array<aws-cdk-lib.IResolvable | aws-cdk-lib.aws_iam.CfnUser.IManagedPolicyRef> | undefined
                ├── 🛑 Failing value is an array
                │      [ 'arn:aws:s3:::bucket1', 'arn:aws:s3:::bucket2' ]
                ╰── 🔍 Failure reason(s):
                    ├─ [as array<aws-cdk-lib.IResolvable | aws-cdk-lib.aws_iam.CfnUser.IManagedPolicyRef>] Index 0: Unable to deserialize value as aws-cdk-lib.IResolvable | aws-cdk-lib.aws_iam.CfnUser.IManagedPolicyRef
                        ├── 🛑 Failing value is a string
                        │      'arn:aws:s3:::bucket1'
                        ╰── 🔍 Failure reason(s):
                            ├─ [as aws-cdk-lib.aws_iam.CfnUser.IManagedPolicyRef] Value is not an object
                            ╰─ [as aws-cdk-lib.IResolvable] Value does not have the "$jsii.byref" key
----

> …it means string slices were passed directly to a property expecting a slice of any.

Reviewer: ```suggestion
…it means string slices were passed directly to a property expecting a slice of any. To solve this, replace `jsii.Strings()` with `jsii.AnyStrings()`. Refer to [this GitHub issue]() for more context and additional solutions.
```
